Adapters/level_zero/queue_profiling.cpp sporadic fail on BMG Win Driver 1.6.34938

### Describe the bug

https://github.com/intel/llvm/actions/runs/18575072524/job/52960382220?pr=14114

```
  # RUN: at line 5
  env UR_L0_DEBUG=-1 env UR_L0_LEAKS_DEBUG=1 SYCL_ENABLE_DEFAULT_CONTEXTS=0 env ONEAPI_DEVICE_SELECTOR=level_zero:gpu  D:\github\_work\llvm\llvm\build-e2e\Adapters\level_zero\Output\queue_profiling.cpp.tmp.out 2>&1 | d:\github\_work\llvm\llvm\install\bin\filecheck.exe --check-prefixes=WITHOUT D:\github\_work\llvm\llvm\llvm\sycl\test-e2e\Adapters\level_zero\queue_profiling.cpp
  # executed command: env UR_L0_DEBUG=-1 env UR_L0_LEAKS_DEBUG=1 SYCL_ENABLE_DEFAULT_CONTEXTS=0 env ONEAPI_DEVICE_SELECTOR=level_zero:gpu 'D:\github\_work\llvm\llvm\build-e2e\Adapters\level_zero\Output\queue_profiling.cpp.tmp.out'
  # executed command: 'd:\github\_work\llvm\llvm\install\bin\filecheck.exe' --check-prefixes=WITHOUT 'D:\github\_work\llvm\llvm\llvm\sycl\test-e2e\Adapters\level_zero\queue_profiling.cpp'
  # .---command stderr------------
  # | D:\github\_work\llvm\llvm\llvm\sycl\test-e2e\Adapters\level_zero\queue_profiling.cpp:14:13: error: WITHOUT: expected string not found in input
  # | // WITHOUT: SYCL exception caught: Profiling information is unavailable as the queue associated with the event does not have the 'enable_profiling' property.
  # |             ^
  # | <stdin>:153:37: note: scanning from here
  # | ze_event_pool_desc_t flags set to: 1
  # |                                     ^
  # | 
  # | Input file: <stdin>
  # | Check file: D:\github\_work\llvm\llvm\llvm\sycl\test-e2e\Adapters\level_zero\queue_profiling.cpp
  # | 
  # | -dump-input=help explains the following input dump.
  # | 
  # | Input was:
  # | <<<<<<
  # |           .
  # |           .
  # |           .
  # |         148: [2025-10-16 14:53:36.896] [ze_loader] [trace] zeCommandListCreateImmediate(hContext, hDevice, altdesc, phCommandList)
  # |  
  # |         149: create command list ordinal: 0, type: immediate, device: 0000022860618D58, inOrder: 1 
  # |         150: [2025-10-16 14:53:36.898] [ze_loader] [trace] zeCommandListCreateImmediate(hContext, hDevice, altdesc, phCommandList)
  # |  
  # |         151: create command list ordinal: 0, type: immediate, device: 0000022860618D58, inOrder: 1 
  # |         152: [2025-10-16 14:53:36.899] [ze_loader] [trace] zeCommandListCreateImmediate(hContext, hDevice, altdesc, phCommandList)
  # |  
  # |         153: ze_event_pool_desc_t flags set to: 1 
  # | check:14                                         X error: no match found
  # |         154: [2025-10-16 14:53:36.900] [ze_loader] [trace] zeEventPoolCreate(hContext, desc, numDevices, phDevicesLocal, phEventPool)
  # |  
  # | check:14     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  # |         155: [2025-10-16 14:53:36.900] [ze_loader] [trace] zeEventCreate(hEventPool, desc, phEvent)
  # |  
  # | check:14     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  # |         156: [2025-10-16 14:53:36.900] [ze_loader] [trace] zeEventCreate(hEventPool, desc, phEvent)
  # |  
  # | check:14     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  # |         157: [2025-10-16 14:53:36.900] [ze_loader] [trace] zeEventCreate(hEventPool, desc, phEvent)
  # |  
  # | check:14     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  # |         158: [2025-10-16 14:53:36.900] [ze_loader] [trace] zeEventCreate(hEventPool, desc, phEvent)
  # |  
  # | check:14     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  # |           .
  # |           .
  # |           .
  # | >>>>>>
  # `-----------------------------
  # error: command failed with exit status: 1
  
```

### To reproduce

_No response_

### Environment

- OS: [e.g Windows/Linux]
- Target device and vendor: [e.g. Intel GPU]
- DPC++ version: [e.g. commit hash or output of `clang++ --version`]
- Dependencies version: [e.g. the output of `sycl-ls --verbose`]


### Additional context

_No response_

--- a/sycl/test-e2e/Adapters/level_zero/queue_profiling.cpp
+++ b/sycl/test-e2e/Adapters/level_zero/queue_profiling.cpp
@@ -1,9 +1,6 @@
 // REQUIRES: gpu, level_zero
 // UNSUPPORTED: ze_debug
 
-// UNSUPPORTED: windows && arch-intel_gpu_bmg_g21
-// UNSUPPORTED-TRACKER: https://github.com/intel/llvm/issues/20385
-
 // RUN: %{build} -o %t.out
 // RUN: env UR_L0_DEBUG=-1 %{l0_leak_check} %{run} %t.out 2>&1 | FileCheck --check-prefixes=WITHOUT %s
 // RUN: env UR_L0_DEBUG=-1 %{l0_leak_check} %{run} %t.out profile 2>&1 | FileCheck --check-prefixes=WITH %s
@@ -14,21 +11,20 @@
 // Check the expected output when queue::enable_profiling is not specified
 //
 // WITHOUT: ze_event_pool_desc_t flags set to: 1
-// WITHOUT: SYCL exception caught: Profiling information is unavailable as the queue associated with the event does not have the 'enable_profiling' property.
 
 // Check the expected output when queue::enable_profiling is specified
 //
 // WITH: ze_event_pool_desc_t flags set to: 5
-// WITH: Device kernel time:
 // clang-format on
 //
 
-#include <iostream>
+#include <cassert>
+#include <cstring>
 #include <sycl/detail/core.hpp>
 #include <sycl/properties/all_properties.hpp>
 using namespace sycl;
 
-int foo(queue &q, int n) {
+void foo(queue &q, int n, bool profiling_enabled) {
   for (int i = 0; i < n; i++) {
 
     sycl::event queue_event = q.submit([&](handler &cgh) {
@@ -44,16 +40,21 @@ int foo(queue &q, int n) {
           sycl::info::event_profiling::command_start>();
       auto endk = queue_event.template get_profiling_info<
           sycl::info::event_profiling::command_end>();
-      auto kernel_time =
-          (float)(endk - startk) * 1e-9f; // to seconds, 1e-6f to milliseconds
-      printf("Device kernel time: %.12fs\n", (float)kernel_time);
-
+      (void)startk;
+      (void)endk;
+      assert(profiling_enabled &&
+             "Expected exception when profiling is not enabled");
     } catch (const sycl::exception &e) {
-      std::cout << "SYCL exception caught: " << e.what() << '\n';
-      return 0;
+      assert(!profiling_enabled &&
+             "Unexpected exception when profiling is enabled");
+      const char *expected_msg =
+          "Profiling information is unavailable as the "
+          "queue associated with the event does not have "
+          "the 'enable_profiling' property.";
+      assert(std::strcmp(e.what(), expected_msg) == 0 &&
+             "Exception message does not match expected message");
     }
   }
-  return n;
 }
 
 int main(int argc, char **argv) {
@@ -67,7 +68,7 @@ int main(int argc, char **argv) {
 
     queue q(gpu_selector_v, propList);
     // Perform the computation
-    foo(q, 2);
+    foo(q, 2, profiling);
   } // SYCL scope
 
   return 0;

